### PR TITLE
Adding support for S3 deployment, along with some other cleanup.

### PR DESCRIPTION
**Pull request by tm0brien**
Some changes included in this PR:

`make build` will now create a package within the `dist/` folder using the hash of its contents as the package suffix. 

`make deploy` will send the built package up to S3. Currently this only handles plain text files, but it will support multiple levels of hierarchy. More work is necessary to support other file types. 

In order for `make deploy` to work, `awskeys.sh` needs to sit at the root of the `usliberties/` directory and contain the appropriate AWS key and secret as `S3KEY` and `S3SECRET`, respectively.

`make serve` will now work to launch the app in a local environment.

Other note: adding the --pretty flag to the Nomplate CLI busts website rendering, so that is currently being omitted in the construction of `index.html`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 lib
 dist
 node_modules
+awskeys.sh
```
Comment (lukebayes) [inline .gitignore:4]: Take a look at your own PRs before merging. (A habit that I've also gotten out of lately!)

File: Makefile [hunk 2]
```diff
@@ -37,13 +42,16 @@ test-w: ${NODE}
 build-module: src/*
 
 serve:
-	$(BABEL_NODE) server.js
+	open dist/index.html
```
Comment (lukebayes) [inline Makefile:45]: :-/ Sorry, we'll need to branch on linux to run `google-chrome` or figure out something better, like script/browser or some such thing...

File: client.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import {renderElement} from 'nomplate';
+
+import constitutional from './src/constitutional';
+import components from './src/components';
+
+if (require.main === module) {
+  global.window.onload = () => {
+    const element = renderElement(constitutional(), global.document);
```
Comment (lukebayes) [inline client.js:8]: I had a realization that we can pull this line out of the onload handler and do half of our work while the browser is waiting for network shizz.

**Comment by lukebayes** [comment]
LGTM - noice!

**Comment by tm0brien** [comment]
thanks! going through this now.

and sorry. i half-remembered linux throughout this :-)

File: .gitignore (hunk 1 above)
Comment (tm0brien) [reply .gitignore:4]: sorry if i'm missing it. what's the question here?

File: Makefile (hunk 2 above)
Comment (tm0brien) [reply Makefile:45]: quick googling suggests `google-chrome` should work for you. i'll add the branching and you can let me know if it works.

**Pull request merged**